From 6879e9859b41001f3d78c35d41990615284353e5 Mon Sep 17 00:00:00 2001
From: Ben Nemec <bnemec@redhat.com>
Date: Wed, 22 Oct 2014 15:13:48 -0500
Subject: [PATCH] Use binary-deps.d for dracut ramdisks

Instead of hard-coding a list of binaries to include in the dracut
ramdisk, use the existing binary-deps.d functionality to provide a
list.  This will allow other ramdisks (such as discovery) to add
the binaries they need.

Change-Id: Ib7ffa15e08db1cc45e93a8f2a5c01369772c93ff
---
 .../dracut-ramdisk/binary-deps.d/dracut-ramdisk    | 15 ++++++++
 .../post-install.d/99-build-dracut-ramdisk         |  2 +-
 .../ramdisk-base/post-install.d/01-ensure-binaries | 45 ++++++++++++++++++++++
 elements/ramdisk/post-install.d/01-ensure-binaries | 45 ----------------------
 4 files changed, 61 insertions(+), 46 deletions(-)
 create mode 100644 elements/dracut-ramdisk/binary-deps.d/dracut-ramdisk
 create mode 100755 elements/ramdisk-base/post-install.d/01-ensure-binaries
 delete mode 100755 elements/ramdisk/post-install.d/01-ensure-binaries

diff --git a/elements/dracut-ramdisk/binary-deps.d/dracut-ramdisk b/elements/dracut-ramdisk/binary-deps.d/dracut-ramdisk
new file mode 100644
index 0000000..ecab2d6
--- /dev/null
+++ b/elements/dracut-ramdisk/binary-deps.d/dracut-ramdisk
@@ -0,0 +1,15 @@
+tail
+head
+awk
+ifconfig
+cut
+expr
+route
+ping
+tgtd
+tgtadm
+nc
+wget
+curl
+tftp
+grep
diff --git a/elements/dracut-ramdisk/post-install.d/99-build-dracut-ramdisk b/elements/dracut-ramdisk/post-install.d/99-build-dracut-ramdisk
index 01e7930..6805f3e 100755
--- a/elements/dracut-ramdisk/post-install.d/99-build-dracut-ramdisk
+++ b/elements/dracut-ramdisk/post-install.d/99-build-dracut-ramdisk
@@ -51,7 +51,7 @@ cp "$TMP_MOUNT_PATH/init" "$MODULE_PATH/80deploy-ramdisk/init.sh"
 # -o: Force omission of these dracut modules.  Our scripts are written for bash,
 #     so dash is not desirable, and plymouth was causing some issues on Ubuntu.
 dracut -N \
-    --install 'tail head awk ifconfig cut expr route ping tgtd tgtadm nc wget curl tftp grep' \
+    --install "$(cat /etc/dib_binary_deps)" \
     --kernel-cmdline "rd.shell rd.debug rd.neednet=1 rd.driver.pre=ahci" \
     --include "$TMP_MOUNT_PATH/init-func" /init-func \
     --kver "${KERNEL_VERSION}" \
diff --git a/elements/ramdisk-base/post-install.d/01-ensure-binaries b/elements/ramdisk-base/post-install.d/01-ensure-binaries
new file mode 100755
index 0000000..41ff4d2
--- /dev/null
+++ b/elements/ramdisk-base/post-install.d/01-ensure-binaries
@@ -0,0 +1,45 @@
+#!/bin/bash
+# Ensure that all binaries listed in ramdisk elements, exist
+
+set -eux
+set -o pipefail
+
+export TARGET_DIR="/tmp/in_target.d/"
+
+if [ -z $TARGET_DIR ] ; then
+    echo "Target dir not specified"
+    exit 1
+fi
+
+if [ ! -d $TARGET_DIR ] ; then
+    echo "Unable to find specified directory in target: $TARGET_DIR"
+    bash
+    exit 1
+fi
+
+BINARY_DEPS=
+
+for _FILE in $(ls ${TARGET_DIR}/binary-deps.d/) ; do
+    _FILE="${TARGET_DIR}/binary-deps.d/${_FILE}"
+    if [ -a $_FILE ]; then
+        for _LINE in $(cat $_FILE) ; do
+            BINARY_DEPS="${BINARY_DEPS} $_LINE"
+        done
+    fi
+done
+
+for _BIN in $BINARY_DEPS ; do
+    _LOCATION=$(which "$_BIN" || echo "")
+    if [ -z "$_LOCATION" ]; then
+        echo "$_BIN is not found in PATH. Please ensure your elements install it"
+        exit 1
+    fi
+done
+
+if [ "$BINARY_DEPS" == "" ]; then
+    echo "No binary-deps found"
+else
+    DEPS_OUTPUT="/etc/dib_binary_deps"
+    echo "Creating binary_deps record at: ${DEPS_OUTPUT}"
+    echo "$BINARY_DEPS" >${DEPS_OUTPUT}
+fi
diff --git a/elements/ramdisk/post-install.d/01-ensure-binaries b/elements/ramdisk/post-install.d/01-ensure-binaries
deleted file mode 100755
index 41ff4d2..0000000
--- a/elements/ramdisk/post-install.d/01-ensure-binaries
+++ /dev/null
@@ -1,45 +0,0 @@
-#!/bin/bash
-# Ensure that all binaries listed in ramdisk elements, exist
-
-set -eux
-set -o pipefail
-
-export TARGET_DIR="/tmp/in_target.d/"
-
-if [ -z $TARGET_DIR ] ; then
-    echo "Target dir not specified"
-    exit 1
-fi
-
-if [ ! -d $TARGET_DIR ] ; then
-    echo "Unable to find specified directory in target: $TARGET_DIR"
-    bash
-    exit 1
-fi
-
-BINARY_DEPS=
-
-for _FILE in $(ls ${TARGET_DIR}/binary-deps.d/) ; do
-    _FILE="${TARGET_DIR}/binary-deps.d/${_FILE}"
-    if [ -a $_FILE ]; then
-        for _LINE in $(cat $_FILE) ; do
-            BINARY_DEPS="${BINARY_DEPS} $_LINE"
-        done
-    fi
-done
-
-for _BIN in $BINARY_DEPS ; do
-    _LOCATION=$(which "$_BIN" || echo "")
-    if [ -z "$_LOCATION" ]; then
-        echo "$_BIN is not found in PATH. Please ensure your elements install it"
-        exit 1
-    fi
-done
-
-if [ "$BINARY_DEPS" == "" ]; then
-    echo "No binary-deps found"
-else
-    DEPS_OUTPUT="/etc/dib_binary_deps"
-    echo "Creating binary_deps record at: ${DEPS_OUTPUT}"
-    echo "$BINARY_DEPS" >${DEPS_OUTPUT}
-fi
